crypto/cryptlib.c

```diff
@@ -413,10 +432,13 @@ int OPENSSL_isservice(void)
 }
 #endif
 
-void OPENSSL_die(const char *message, const char *file, int line)
+void OPENSSL_die2(const char *message, const char *file, int line, const char *func)
```

Comment:
Is adding a new public API function justified for this? If we were to add it, it should be called `OPENSSL_die_ex`, and should come with documentation in the pod files.

Reply:
`OPENSSL_die` is already public-but-undocumented so I didn't see it as a major change. Not entirely sure _why_ it's public tbh (along with `OPENSSL_assert`); should those be deprecated from the public API?

Reply:
Perhaps they should be deprecated. I don't see a need to add any *more* functions like this. I'm also not really sure why we need it anyway. It seems somewhat tangential to the main point of this PR.

Reply:
Perhaps more importantly, OpenSSL *removed all internal function codes* from its error stack.  Adding this back seems wrong.

Reply:
Er, `func` here has nothing to do with the internal function codes, and everything to do with compiler generated function *names*, which we do use instead of the old function codes when available.

I have told you this before, @richsalz

Reply:
But functiona names are not currently used in the assertion stuff.

Reply:
And? That is of course a choice, but I'd rather not factor in facts that have no relevance here.

Reply:
We disagree.

It doesn't seem right to add a new undocumented function, tho.

Reply:
Isn't it against policy to add new undocumented functions?

Reply:
I'd agree that this should be internal, I'm just not sure how to deal with the existing public macro (nor how to deprecate things in general).